### ISSUE TITLE
Invoke multiple storage.stores in parallel

### ISSUE THREAD

**Issue by pjenvey** [mozilla-services/contile#232]
We could probably `join!` the potentially multiple calls to `StorageImage::store` to invoke them all in parallel. This may not be a huge win since there's currently a limit of 2 tiles and some of the work is CPU based (image hashing), but there's also at least one and potentially two http requests per store call.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/DISCO-1170)
┆Epic: Contile: maintenance

**Comment by jrconlin** [comment]
Honestly, I wonder if we should include live profiling so that we're not just guessing about some of these optimizations. I agree that we could do that, but I'm guessing that normally it'd be only one or two threads on a given server that might have to do this operation once every 15 days or so as new images are uploaded. (Of course, that goes out the window if our partner goes nuts trying out new images live.)

**Comment by pjenvey** [comment]
Agreed this is low priority, I just wanted an issue down before I forgot all about it.

The upload will definitely be rare but the (usually) pair of fetch/check calls could take advantage of it.
